### PR TITLE
Disabled leagcy engine build: Disable code related to ambigs.

### PR DESCRIPTION
**Pull request by amitdo**

### CODE REVIEW

**Comment by amitdo** [comment]
**Please don't merge yet!**

**Comment by amitdo** [comment]
It was compiled fine with --disable-shared.

With --disable-static it gives me a linker error.

**Comment by amitdo** [comment]
>/usr/bin/ld: ../textord/.libs/libtesseract_textord.a(cjkpitch.o): relocation R_X86_64_PC32 against symbol `_ZN13GenericVectorIfE5clearEv' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
Makefile:540: recipe for target 'libtesseract.la' failed

**Comment by amitdo** [comment]
@stweil, can you help me with the linker error?

**Comment by stweil** [comment]
It works for me without a linker error (Debian stable, g++).

File: src/ccutil/Makefile.am [hunk 1]
```diff
@@ -10,6 +10,10 @@ AM_CXXFLAGS += -fvisibility=hidden -fvisibility-inlines-hidden
 AM_CPPFLAGS += -DTESS_EXPORTS
 endif
 
+if DISABLED_LEGACY_ENGINE
+AM_CPPFLAGS += -DDISABLED_LEGACY_ENGINE
+endif
+
```
Comment (stweil) [inline src/ccutil/Makefile.am]: This is required in `textord`, too.

File: src/ccutil/ambigs.h [hunk 2]
```diff
@@ -215,4 +217,19 @@ class UnicharAmbigs {
 
 }  // namespace tesseract
 
+
+#else
```
Comment (stweil) [inline src/ccutil/ambigs.h]: The `#else` part looks like a hack which can be avoided.

File: src/ccutil/Makefile.am (hunk 1 above)
Comment (stweil) [reply src/ccutil/Makefile.am]: ... and for `lstm` as well. Done in my additional commit.

File: src/ccutil/ambigs.h (hunk 2 above)
Comment (stweil) [reply src/ccutil/ambigs.h]: I removed it in my additional commit.

Comment (stweil) [reply src/ccutil/ambigs.h]: @amitdo, do we already have more hacks like that? Then they should be examined closer, too.

File: src/ccutil/Makefile.am (hunk 1 above)
Comment (amitdo) [reply src/ccutil/Makefile.am]: > This is required in `textord`, too.

Before asking for your help, I tried adding that code to textord, but there was still a linker error.

> ... and for `lstm` as well.

This one (together with the others) solves the issue.

File: src/ccutil/ambigs.h (hunk 2 above)
Comment (amitdo) [reply src/ccutil/ambigs.h]: Yes, that code was meant to be temporary.

My plan was to fix it later. You already fixed it yourself. Thanks!

**Comment by amitdo** [comment]
The PR is ready to be merged now.

**Comment by stweil** [comment]
Thank you. It is merged now.

**Pull request merged**